plugins/config_manager.py

```diff
@@ -0,0 +1,241 @@
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from dnfpluginscore import _, logger
+
+import dnf
+import dnf.cli
+import dnfpluginscore
+import dnfpluginscore.lib
+import os
+import re
+import urlparse
+
+
+class ConfigManager(dnf.Plugin):
+
+    name = 'config-manager'
+
+    def __init__(self, base, cli):
+        super(ConfigManager, self).__init__(base, cli)
+        self.base = base
+        self.cli = cli
+        if self.cli is not None:
+            self.cli.register_command(ConfigManagerCommand)
+
+
+class ConfigManagerCommand(dnf.cli.Command):
+
+    aliases = ['config-manager']
+    summary = _('manage dnf configuration options and repositories')
+
+    def __init__(self, cli):
+        super(ConfigManagerCommand, self).__init__(cli)
+        self.opts = None
+        self.parser = None
+
+    def configure(self, args):
+        # setup sack and populate it with enabled repos
+        demands = self.cli.demands
+        demands.available_repos = True
+
+        self.parser = dnfpluginscore.ArgumentParser(self.aliases[0])
+        self.parser.add_argument('repo', nargs='*',
+                    help=_('repo to modify'))
+        self.parser.add_argument('--save',
+                    default=False, action='store_true',
+                    help=_('save the current options (useful with --setopt)'))
+        self.parser.add_argument('--set-enabled',
+                    default=False, action='store_true',
+                    help=_('enable the specified repos (automatically saves)'))
+        self.parser.add_argument('--set-disabled',
+                    default=False, action='store_true',
+                    help=_('disable the specified repos (automatically saves)'))
+        self.parser.add_argument('--add-repo',
+                    default=[], action='append', metavar='URL',
+                    help=_('add (and enable) the repo from the specified file or url'))
+
+        self.opts = self.parser.parse_args(args)
+
+        if self.opts.help_cmd:
+            print(self.parser.format_help())
+            return
+
+        if (self.opts.save or self.opts.set_enabled or
+                self.opts.set_disabled or self.opts.add_repo):
+            self.cli.demands.root_user = True
+
+
+    def run(self, _):
+        """Execute the util action here."""
+
+        if self.opts.set_enabled and self.opts.set_disabled:
+            logger.error(_("Error: Trying to enable and disable repos at the same time."))
```

Comment:
sure, _ is a bug - originally it was _args and I wanted to silent pylint's complains so it ended up as simple _ :-)